python/ray/_private/runtime_env.py

```diff
@@ -340,27 +345,27 @@ def upload_runtime_env_package_if_needed(job_config: JobConfig) -> None:
             logger.info(f"{pkg_uri} has been pushed with {pkg_size} bytes")
 
 
-def ensure_runtime_env_setup(pkg_uris: List[str]) -> None:
+def ensure_runtime_env_setup(pkg_uris: List[str]) -> str:
     """Make sure all required packages are downloaded it local.
 
     Necessary packages required to run the job will be downloaded
     into local file system if it doesn't exist.
 
     Args:
         pkg_uri list(str): Package of the working dir for the runtime env.
-    """
 
+    Return:
+        Working directory is returned.
+    """
+    pkg_dir = None
     assert _internal_kv_initialized()
     for pkg_uri in pkg_uris:
-        pkg_file = Path(_get_local_path(pkg_uri))
         # For each node, the package will only be downloaded one time
         # Locking to avoid multiple process download concurrently
+        pkg_file = Path(_get_local_path(pkg_uri))
         with FileLock(str(pkg_file) + ".lock"):
-            # TODO(yic): checksum calculation is required
-            if pkg_file.exists():
-                logger.debug(
-                    f"{pkg_uri} has existed locally, skip downloading")
-            else:
-                pkg_size = fetch_package(pkg_uri, pkg_file)
-                logger.debug(f"Downloaded {pkg_size} bytes into {pkg_file}")
-        sys.path.insert(0, str(pkg_file))
+            pkg_dir = fetch_package(pkg_uri)
+        sys.path.insert(0, str(pkg_dir))
+    # Right now, multiple pkg_uris are not supported correctly.
+    # We return the last one as working directory
```

Comment:
We plan to move this to a separate field, right? This seems like it could cause confusion.

Reply:
Oh, not really. It has been moved to a separate file in `runtime_env`. Here it's for multiple envs, like the current env is too big, and we need to split them. In this case, we need to create a temp dir and link everything there. Right now, since we don't do that, we only have one uri.

Reply:
Gotcha, that makes sense.